File: src/main/java/services/FileService.java
package services;

import dao.FileDao;
import models.File;

import java.util.List;

public class FileService extends Service<File> {

    private FileDao fileDao;

    public FileService(int userID, String password) {
        fileDao = new FileDao(userID, password);
    }

    public File find(int id) {
        return fileDao.findById(id);
    }

    public void save(File file) {
        fileDao.save(file);
    }

    public void update(File file) {
        fileDao.update(file);
    }

    public void delete(File file) {
        fileDao.delete(file);
    }

    public List<File> getAll() {
        return fileDao.getAll();
    }
}

File: src/main/kotlin/services/FileService.kt
package services

import dao.FileDao
import models.File

class FileService(userID: Int, password: String) : Service<File>() {

    private val fileDao: FileDao = FileDao(userID, password)

    override fun find(id: Int): File? {
        return fileDao.findById(id)
    }

    fun find(name: String, path: String) : File? {
        return fileDao.findByNamePath(name, path)
    }

    override fun save(obj: File) {
        if (fileDao.findByNamePath(obj.name, obj.path!!) == null) {
            fileDao.save(obj)
        }
    }

    override fun update(obj: File) {
        fileDao.update(obj)
    }

    override fun delete(obj: File) {
        fileDao.delete(obj)
    }

    override fun getAll(): List<File> {
        return fileDao.all
    }
}
